### PR TITLE
Add JNI support for apply_boolean_mask

### PR DESCRIPTION
**Pull request by res-life**
Contributes to #10650

Add JNI support for `apply_boolean_mask`

Refer to the descriptions of PR #10773

Signed-off-by: Chong Gao <res_life@163.com>

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

File: java/src/main/java/ai/rapids/cudf/ColumnView.java [hunk 1]
```diff
@@ -3498,6 +3498,10 @@ public final Scalar getScalarElement(int index) {
     return new Scalar(getType(), getElement(getNativeView(), index));
   }
 
+  public final ColumnVector applyBooleanMask(ColumnView booleanMaskView) {
+    return new ColumnVector(applyBooleanMask(getNativeView(), booleanMaskView.getNativeView()));
```
Comment (firestarman) [inline java/src/main/java/ai/rapids/cudf/ColumnView.java:3502]: Here can add an assertion for the types of the input and this before calling into the native.

Comment (firestarman) [reply java/src/main/java/ai/rapids/cudf/ColumnView.java:3502]: Better to add the description for this public API, you may simply copy it from the native API's doc.

**Review by firestarman — APPROVED** [review]

**Comment by firestarman** [comment]
rerun tests

File: java/src/main/native/src/ColumnViewJni.cpp [hunk 2]
```diff
@@ -2223,4 +2224,24 @@ JNIEXPORT jlongArray JNICALL Java_ai_rapids_cudf_ColumnView_repeatStringsSizes(
...
+  try {
+    cudf::jni::auto_set_device(env);
+
+    cudf::column_view *list_column = reinterpret_cast<cudf::column_view *>(list_column_handle);
+    cudf::lists_column_view list_view = cudf::lists_column_view(*list_column);
+
+    cudf::column_view *boolean_mask_list_column =
+        reinterpret_cast<cudf::column_view *>(boolean_mask_list_column_handle);
+    cudf::lists_column_view boolean_mask_list_view =
+        cudf::lists_column_view(*boolean_mask_list_column);
```
Comment (jlowe) [inline java/src/main/native/src/ColumnViewJni.cpp:2240]: const nits:
```suggestion
    cudf::column_view const *list_column = reinterpret_cast<cudf::column_view const *>(list_column_handle);
    cudf::lists_column_view const list_view = cudf::lists_column_view(*list_column);

    cudf::column_view const *boolean_mask_list_column =
        reinterpret_cast<cudf::column_view const *>(boolean_mask_list_column_handle);
    cudf::lists_column_view const boolean_mask_list_view =
        cudf::lists_column_view(*boolean_mask_list_column);
```

**Review by jlowe — APPROVED** [review]
const nits but otherwise lgtm.

**Comment by res-life** [comment]
rerun tests

File: java/src/main/native/src/ColumnViewJni.cpp (hunk 2 above)
Comment (res-life) [reply java/src/main/native/src/ColumnViewJni.cpp]: Done.

**Comment by res-life** [comment]
@gpucibot merge

**Comment by res-life** [comment]
ok to test

**Comment by firestarman** [comment]
ok to test

**Comment by harrism** [comment]
ok to test

**Comment by res-life** [comment]
@gpucibot merge

**Comment by harrism** [comment]
@gpucibot merge

**Pull request merged**